dockstore-webservice/src/main/java/io/dockstore/webservice/core/Version.java

```diff
@@ -273,6 +287,9 @@ public void setName(String name) {
     public void updateVerified(boolean newVerified, String newVerifiedSource) {
         this.getVersionMetadata().verified = newVerified;
         this.getVersionMetadata().verifiedSource = newVerifiedSource;
+        // Alternate solution
+        this.getVersionMetadata().verified = this.getSourceFiles().stream().anyMatch(file -> file.getVerifiedBySource().values().stream().anyMatch(innerEntry -> innerEntry.verified));
```

Comment:
Maybe you had the same thought, I think I prefer this but extended to cache what `getVerifiedSource` calculates.

(and a migration or manual cleanup for inconsistent content)